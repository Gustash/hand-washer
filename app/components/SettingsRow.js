import React from 'react'
import { TouchableOpacity, Text, StyleSheet } from 'react-native'
import { colors } from 'app/config'

const SettingsRow = ({ text, rightElement, style, onPress }) => (
  <TouchableOpacity
    style={[styles.container, style]}
    onPress={onPress}
    disabled={!onPress}
  >
    <Text testId="settings-row-text" style={styles.text}>
      {text}
    </Text>
    {rightElement}
  </TouchableOpacity>
)

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    borderBottomColor: colors.separator,
    borderBottomWidth: 1,
    paddingVertical: 12,
    paddingLeft: 8,
    paddingRight: 27,
    alignItems: 'center',
  },
  text: {
    color: colors.darkGrey,
    fontSize: 17,
    flex: 1,
  },
})

export default SettingsRow
