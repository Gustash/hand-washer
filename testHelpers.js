export const findNodeWithTestId = (wrapper, testId) =>
  wrapper.findWhere(node => node.prop('testId') === testId)
