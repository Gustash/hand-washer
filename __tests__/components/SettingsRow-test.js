import React from 'react'
import { Text, StatusBar, TouchableOpacity } from 'react-native'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import SettingsRow from '../../app/components/SettingsRow'
import { findNodeWithTestId } from '../../testHelpers'

const mockProps = props => ({
  text: 'Text',
  rightElement: <Text>Right Element</Text>,
  style: { marginTop: 1000 },
  onPress: jest.fn(),
  ...props,
})

describe('<SettingsRow />', () => {
  let props
  let wrapper

  beforeEach(() => {
    props = mockProps()
    wrapper = shallow(<SettingsRow {...props} />)
  })

  it('matches snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('applies text', () => {
    expect(
      findNodeWithTestId(wrapper, 'settings-row-text').prop('children')
    ).toBe('Text')

    wrapper.setProps({
      ...props,
      text: 'Another Text',
    })

    expect(
      findNodeWithTestId(wrapper, 'settings-row-text').prop('children')
    ).toBe('Another Text')
  })

  it('applies right element', () => {
    expect(wrapper.containsMatchingElement(<Text>Right Element</Text>)).toBe(
      true
    )

    wrapper.setProps({
      ...props,
      rightElement: <StatusBar backgroundColor="#fefefe" />,
    })

    expect(
      wrapper.containsMatchingElement(<StatusBar backgroundColor="#fefefe" />)
    ).toBe(true)
  })

  it('applies style', () => {
    expect(wrapper.first().prop('style')).toContainEqual({
      marginTop: 1000,
    })

    wrapper.setProps({
      ...props,
      style: {
        marginBottom: 1000,
      },
    })

    expect(wrapper.first().prop('style')).toContainEqual({
      marginBottom: 1000,
    })
  })

  it('disables touchable without onPress', () => {
    expect(wrapper.find(TouchableOpacity).prop('disabled')).toBe(false)

    wrapper.setProps({
      ...props,
      onPress: null,
    })

    expect(wrapper.find(TouchableOpacity).prop('disabled')).toBe(true)
  })

  it('calls onPress', () => {
    expect(props.onPress).toHaveBeenCalledTimes(0)

    wrapper.simulate('press')

    expect(props.onPress).toHaveBeenCalledTimes(1)
  })
})
