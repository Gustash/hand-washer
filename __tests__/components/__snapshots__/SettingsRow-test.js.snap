// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsRow /> matches snapshot 1`] = `
<ForwardRef
  disabled={false}
  onPress={[MockFunction]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomColor": "#E6E6E6",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "paddingLeft": 8,
        "paddingRight": 27,
        "paddingVertical": 12,
      },
      Object {
        "marginTop": 1000,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#4A4A4A",
        "flex": 1,
        "fontSize": 17,
      }
    }
    testId="settings-row-text"
  >
    Text
  </Text>
  <Text>
    Right Element
  </Text>
</ForwardRef>
`;
